ParProcCo/job_scheduler.py

```diff
@@ -183,199 +391,192 @@ def _create_template(self, i: int) -> drmaa2.JobTemplate:
         )
-        logging.info(
-            f"creating template with jobscript: {str(self.jobscript)} and args: {args}"
+        self.jobscript_command = " ".join(
+            [f"#!/bin/bash\n{self.jobscript_path}", *args]
         )
-
-        self.resources["m_mem_free"] = self.memory
-        jt = drmaa2.JobTemplate(
-            {
-                "job_name": self.job_name,
-                "job_category": self.project,
-                "remote_command": str(self.jobscript),
-                "min_slots": self.cores,
-                "args": args,
-                "resource_limits": self.resources,
-                "working_directory": str(self.working_directory),
-                "output_path": stdout_fp,
-                "error_path": stderr_fp,
-                "queue_name": self.queue,
-                "implementation_specific": {
-                    "uge_jt_pe": "smp",
-                },
-            }
+        logging.info(f"creating submission with command: {self.jobscript_command}")
+        job = JobProperties(
+            name=self.job_name,
+            partition=self.partition,
+            cpus_per_task=self.cores,
+            environment=self.job_env,
+            memory_per_cpu=self.memory,
+            current_working_directory=str(self.working_directory),
+            standard_output=stdout_fp,
+            standard_error=stderr_fp,
+            get_user_environment="10L",
         )
-        job_env = dict(self.job_env) if self.job_env is not None else {}
-        test_ppc_dir = os.getenv("TEST_PPC_DIR")
-        if test_ppc_dir:
-            logging.debug("Adding TEST_PPC_DIR=%s to environment", test_ppc_dir)
-            job_env.update(TEST_PPC_DIR=test_ppc_dir)
-        jt.job_environment = job_env
-        return jt
 
-    def _wait_for_jobs(self, session: drmaa2.JobSession) -> None:
+        return JobSubmission(script=self.jobscript_command, job=job)
+
+    def fetch_and_update_state(self, job_id: int) -> JobResponseProperties:
+        ji = self.get_job(job_id)
+        self.update_status_infos(ji)
+        return ji
+
+    def wait_all_jobs_terminated(
+        self, job_ids: List[int], check_time: int
+    ) -> list[int]:
+        wait_until = time.time() + check_time
+        remaining_jobs = job_ids.copy()
+        sleep_time = min(60, check_time)
+        while len(remaining_jobs) > 0 and time.time() <= wait_until:
+            for job_id in list(remaining_jobs):
+                self.fetch_and_update_state(job_id)
+                if self.status_infos[job_id].current_state in STATEGROUP.ENDED:
+                    remaining_jobs.remove(job_id)
+            if len(remaining_jobs) > 0:
+                time.sleep(sleep_time)
+        return remaining_jobs
+
+    def wait_all_jobs_started(self, job_ids: List[int], check_time: int) -> list[int]:
+        wait_until = time.time() + check_time
+        remaining_jobs = job_ids.copy()
+        while len(remaining_jobs) > 0 and time.time() <= wait_until:
```

Comment:
Maybe add a `sleep_time` so we don't hammer the endpoint. Also these two functions are very similar to make a `wait_all_jobs(.., required_state)`